### PR TITLE
slweb: init at 0.5

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

slweb is a static website generator which aims at being simplistic. It transforms custom Markdown-like syntax into HTML.
Homepage: https://strahinja.srht.site/slweb/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/slweb/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv
+, fetchgit
+, redo-apenwarr
+}:
+
+stdenv.mkDerivation rec {
+  pname = "slweb";
+  version = "0.5";
+
+  src = fetchgit {
```
Comment (deejayem) [inline pkgs/applications/misc/slweb/default.nix:10]: Shouldn't this use `fetchFromSourcehut`?

File: pkgs/applications/misc/slweb/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    redo install
+    runHook postInstall
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    description = "A static website generator which aims at being simplistic";
+    homepage = "https://strahinja.srht.site/slweb/";
+    license = licenses.gpl3;
```
Comment (deejayem) [inline pkgs/applications/misc/slweb/default.nix:30]: I think this should be `licenses.gpl3Plus`

File: pkgs/applications/misc/slweb/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/applications/misc/slweb/default.nix]: Oh ! I didn't know about this ! Done :)

File: pkgs/applications/misc/slweb/default.nix (hunk 2 above)
Comment (GaetanLepage) [reply pkgs/applications/misc/slweb/default.nix]: Done

**Comment by GaetanLepage** [comment]
> In case you haven't noticed, one of ofborg's checks failed: slweb.passthru.tests on x86_64-darwin

Indeed. However, I do not own a mac and I do not really know what I should do...
Any idea ?

**Comment by deejayem** [comment]
> > In case you haven't noticed, one of ofborg's checks failed: slweb.passthru.tests on x86_64-darwin
> 
> Indeed. However, I do not own a mac and I do not really know what I should do... Any idea ?

I think setting platforms to `platforms.linux` for now should sort it. (I'm not sure of the pros and cons of that vs setting `broken = stdenv.isDarwin;`)

**Comment by deejayem** [comment]
Result of `nixpkgs-review pr 205314` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>slweb</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
It seems to be OK to merge right ?

**Review by deejayem — APPROVED** [review]
Built and tested on `aarch64-linux`. The diff looks good to me.

File: pkgs/applications/misc/slweb/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/slweb/default.nix:1]:
```suggestion
{ lib
, stdenv
```

File: pkgs/applications/misc/slweb/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-GoDumiysqIkWj2HTPQv2gheYsf4fWjtCNPFS/1R0tzc=";
+  };
+
+  nativeBuildInputs = [ redo-apenwarr ];
+
+  installPhase = ''
+    runHook preInstall
+    export PREFIX=$out
+    redo install
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/slweb/default.nix:22]:
```suggestion
    PREFIX=$out redo install
```

Comment (GaetanLepage) [reply pkgs/applications/misc/slweb/default.nix]: Done !

File: pkgs/applications/misc/slweb/default.nix (hunk 3 above)
Comment (GaetanLepage) [reply pkgs/applications/misc/slweb/default.nix]: Done :)

**Pull request merged**